apps/web/src/app/(private)/statistiques/page.tsx

```diff
@@ -8,15 +8,23 @@ import { FamilySituationChart } from '@mss/web/app/(private)/statistiques/Family
 import { AgeChart } from '@mss/web/app/(private)/statistiques/AgeChart'
 import { ChartJs } from '@mss/web/app/(private)/statistiques/ChartJs'
 import { Routes } from '@mss/web/app/routing/routes'
+import { canAccessStatsPage } from '@mss/web/security/rules'
+import { notFound } from 'next/navigation'
 
 const StatistiquesPage = async () => {
-  const { structureId } = await getAuthenticatedAgent()
+  const user = await getAuthenticatedAgent()
+
+  if (!canAccessStatsPage(user)) {
+    notFound()
+    return null
```

Comment:
J'ai TypeScript qui me met une info (rien de bloquant) comme quoi ce code est inatteignable. Ça te parle ?

Reply:
Unreachable code dans un bloc conditionnel, c'est si la condition du `if` est considérée / inférée comme étant toujours `false`

La ca serait le cas si Typescript a inféré que `canAccessStatsPage()` return toujours `true`

Vu le code on dirait que ca a été vrai un moment en dev, et plus maintenant, essaye de restart le Typescript service :)

Reply:
Pardon, le code inatteignable n'est pas le bloc conditionnel mais uniquement la ligne du `return`, comme si `notFound()` forçait le retour de la fonction :

![Code inatteignable](https://wtf.roflcopter.fr/pics/uU2bXF7C/liZwLyj7.png)
![Code inatteignable](https://wtf.roflcopter.fr/pics/rS5jwpzR/ryTwYvWV.png)